bin/custom_lint.dart

```diff
@@ -1,23 +1,51 @@
+import 'dart:core';
 import 'dart:isolate';
 
 import 'package:analyzer/dart/analysis/results.dart';
-import 'package:analyzer/dart/ast/ast.dart';
-import 'package:architecture_linter/src/layer_rule.dart';
+import 'package:architecture_linter/src/configuration_reader/configuration_reader.dart';
+import 'package:architecture_linter/src/extensions/string_extensions.dart';
+import 'package:architecture_linter/src/project_name_reader/project_name_reader.dart';
 import 'package:custom_lint_builder/custom_lint_builder.dart';
+import 'package:architecture_linter/src/configuration/layer.dart';
 
-void main(List<String> args, SendPort sendPort) {
-  startPlugin(sendPort, ArchitectureLinter());
+void main(List<String> args, SendPort port) {
+  startPlugin(port, ArchitectureLinter());
 }
 
 class ArchitectureLinter extends PluginBase {
+  final configReader = ConfigurationReader();
+  final projectNameReader = ProjectNameReader();
+
   @override
   Stream<Lint> getLints(ResolvedUnitResult resolvedUnitResult) async* {
-    final currentPath = resolvedUnitResult.path;
-    final layerRuleList = _getLayerRuleList();
+    final analysis = resolvedUnitResult;
+    final components = analysis.libraryElement.entryPoint?.location?.components;
+    final rootProjectName = projectNameReader.readRootName(components);
+
+    if (rootProjectName.isEmpty) return;
+
+    final path = analysis.path; // Absolute path for analyzed file
+    final packagePath = path.trimTo(rootProjectName); // Absolute path from start to `rootProjectName`
+    final libraryPath = "$packagePath/lib";
+
+    if (!path.startsWith(libraryPath)) {
+      return;
+    }
+
+    final config = await configReader.readConfiguration(packagePath, "architecture.yaml");
```

Comment:
Jako, że będziemy raczej rezygnować z libki, to nie trzeba zmieniać teraz
Ale chyba trzeba by było pomyśleć nad optymalniejszym czytaniem konfigu, teraz by czytało ten konfig per plik
Wdyt?

Reply:
I tak poprawiłem inicjalizacje na lazy bo to było głupie (czyt. nieoptymalne)